### PR TITLE
Create user profile type changes

### PR DESCRIPTION
**Pull request by verdefred**
## Summary
We want to have more control over when the users change their profile type or when someone on the admin team approves someone. For this we're creating a new model that tracks that

### CODE REVIEW

File: app/models/user_profile_type_change.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class UserProfileTypeChange < ApplicationRecord
+  belongs_to :user, class_name: "User"
+  belongs_to :who_dunnit, class_name: "User", optional: true
```
Comment (RubenSousaDinis) [inline app/models/user_profile_type_change.rb:3]: Would it ever happen that a change is done and the who_dunnit is null?

File: app/services/api/update_talent.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 class API::UpdateTalent
-  attr_reader :talent, :success
+  attr_reader :talent, :current_user, :success
```
Comment (RubenSousaDinis) [inline app/services/api/update_talent.rb:2]: Personal preference, but in the business area of the application (services) we shouldn't have the current_user notion. It should only live in the requests area. Here it should simply be a user.

File: db/migrate/20220609162621_create_user_profile_type_changes.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class CreateUserProfileTypeChanges < ActiveRecord::Migration[6.1]
+  def change
+    create_table :user_profile_type_changes do |t|
+      t.integer :user_id
```
Comment (RubenSousaDinis) [inline db/migrate/20220609162621_create_user_profile_type_changes.rb:4]: we're not creating  a foreign_key with this approach. Also it should be not nullable.

File: spec/services/users/update_profile_type.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+
+RSpec.describe Users::UpdateProfileType do
+  include ActiveJob::TestHelper
```
Comment (RubenSousaDinis) [inline spec/services/users/update_profile_type.rb:4]: not needed

File: spec/services/users/update_profile_type.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      expect(result.user_id).to be(user.id)
+      expect(result.who_dunnit_id).to be(user.id)
+      expect(result.new_profile_type).to eq(new_profile_type)
+    end
+  end
+
+  context "when an invalid profile_type is provided" do
+    let(:new_profile_type) { "gibberish" }
+
+    it "it updates the profile_type" do
```
Comment (RubenSousaDinis) [inline spec/services/users/update_profile_type.rb:36]:
```suggestion
    it "it fails to update the profile_type" do
```

**Review by RubenSousaDinis — COMMENTED** [review]
Some comments to be considered.

File: app/services/api/update_talent.rb (hunk 2 above)
Comment (verdefred) [reply app/services/api/update_talent.rb:2]: makes sense, I'll make the change

**Review by RubenSousaDinis — APPROVED** [review]

**Review by RubenSousaDinis — APPROVED** [review]

**Pull request merged**